code/messaging/src/main/java/com/adobe/marketing/mobile/PropositionItem.java

```diff
@@ -84,10 +95,10 @@ public String getContent() {
     /**
      * Gets the {@code Proposition} reference.
      *
-     * @return {@link Proposition} instance {@link WeakReference}.
+     * @return {@link Proposition} instance {@link SoftReference}.
      */
-    public Proposition getProposition() {
-        return proposition.get();
+    public SoftReference<Proposition> getPropositionReference() {
```

Comment:
We need a getter only for the object for a customer to access the parent proposition
```
public Proposition getProposition() {
    return propositionReference.get();
}
```